botorch/utils/sampling.py

```diff
@@ -830,6 +830,32 @@ def draw(self, n: int = 1, seed: Optional[int] = None) -> Tensor:
         return samples
 
 
+def normalize_linear_constraints(
+    bounds: Tensor, constraints: List[Tuple[Tensor, Tensor, float]]
+) -> List[Tuple[Tensor, Tensor, float]]:
+    r"""Normalize linear constraints to the unit cube.
+
+    Args:
+        bounds (Tensor): A `2 x d`-dim tensor containing the box bounds.
+        constraints (List[Tuple[Tensor, Tensor, float]]): A list of
+            tuples (indices, coefficients, rhs), with each tuple encoding
+            an inequality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) >= rhs` or
+            `\sum_i (X[indices[i]] * coefficients[i]) = rhs`.
+    """
+
+    new_constraints = []
+    for index, coefficient, rhs in constraints:
+        # the index is cast to int64 since in some tests the index is
+        # given as a indexing incompatible dtype
```

Comment:
It looks like you fixed this in the tests also? Can we get rid of this comment and the casting below then? I'd like to avoid a situation where someone is confusing the order or the `indices` and `coefficient` tensors and suddenly ends up with something that doesn't error out but produces garbage results.

Reply:
There was another wrong type definition in one of the intializer tests. I also corrected this and removed the explicit cast.